packages/medusa/src/api/routes/admin/orders/request-return.ts

```diff
@@ -89,176 +90,177 @@ export default async (req, res) => {
 
   const headerKey = req.get("Idempotency-Key") || ""
 
-  let idempotencyKey
-  try {
-    idempotencyKey = await idempotencyKeyService.initializeRequest(
-      headerKey,
-      req.method,
-      req.params,
-      req.path
-    )
-  } catch (error) {
-    res.status(409).send("Failed to create idempotency key")
-    return
-  }
-
-  res.setHeader("Access-Control-Expose-Headers", "Idempotency-Key")
-  res.setHeader("Idempotency-Key", idempotencyKey.idempotency_key)
-
-  try {
-    const orderService: OrderService = req.scope.resolve("orderService")
-    const returnService: ReturnService = req.scope.resolve("returnService")
-    const eventBus: EventBusService = req.scope.resolve("eventBusService")
-
-    let inProgress = true
-    let err = false
-
-    while (inProgress) {
-      switch (idempotencyKey.recovery_point) {
-        case "started": {
-          const { key, error } = await idempotencyKeyService.workStage(
-            idempotencyKey.idempotency_key,
-            async (manager) => {
-              const returnObj: ReturnObj = {
-                order_id: id,
-                idempotency_key: idempotencyKey.idempotency_key,
-                items: value.items,
-              }
-
-              if (value.return_shipping) {
-                returnObj.shipping_method = value.return_shipping
-              }
-
-              if (typeof value.refund !== "undefined" && value.refund < 0) {
-                returnObj.refund_amount = 0
-              } else {
-                if (value.refund && value.refund >= 0) {
-                  returnObj.refund_amount = value.refund
-                }
-              }
+  const manager: EntityManager = req.scope.resolve("manager")
+  await manager.transaction(async (transactionManager) => {
+    let idempotencyKey
+    try {
+      idempotencyKey = await idempotencyKeyService
+        .withTransaction(transactionManager)
+        .initializeRequest(headerKey, req.method, req.params, req.path)
+    } catch (error) {
+      res.status(409).send("Failed to create idempotency key")
+      return
+    }
```

Comment:
The `idempotencyKeyService.workStage` creates a transaction so when doing it this way you are creating nested transactions which we don't want.

Reply:
no, the workStage, from what I see, reuse the given transaction and pass it the `func`. Since the manager from the atonicPhase is the one given through the `withTransction`

Reply:
Why do we want it to happen in a separate transaction than the one currently running?